### PR TITLE
batt 0.3.1

### PR DESCRIPTION
**Pull request by charlie0129**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change? https://github.com/Homebrew/homebrew-core/pull/193169 exists but should prefer this one (see why below)
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

[batt](https://github.com/charlie0129/batt) v0.3.1 has some Homebrew-specific changes that will make this formulae less hacky.

This PR has some changes in the formulae to have batt integrate better with Homebrew:

- No source-editing during building (made possible by upstream changes).
- Add brew-specific batt building options, i.e. remove built-in install/uninstall commands in batt cli.
- Better handling of config locations.

### CODE REVIEW

File: Formula/b/batt.rb [hunk 1]
```diff
@@ -17,27 +17,27 @@ class Batt < Formula
...
-    # talk to anything non-root. Changing the defaults here.
-    inreplace "conf.go", "AllowNonRootAccess:      false,",
-                         "AllowNonRootAccess:      true,"
-    # Limit config path to Homebrew prefix. The socket remains in /var/run
-    # to deter non-root invocation of daemon
-    inreplace ["conf.go", "README.md"], "/etc/batt.json", etc/"batt.json"
-    # Due to local changes version tag would show v0.2.1-dirty
-    system "make", "VERSION=v#{version}"
+    # Point to the correct path for the binary
     inreplace "hack/cc.chlc.batt.plist", "/path/to/batt", bin/"batt"
```
Comment (carlocab) [inline Formula/b/batt.rb:21]: We should probably point the `plist` at a stable path:
```suggestion
    inreplace "hack/cc.chlc.batt.plist", "/path/to/batt", opt_bin/"batt"
```

File: Formula/b/batt.rb [hunk 2]
```diff
@@ -17,27 +17,27 @@ class Batt < Formula
...
-    system "plutil", "-insert", "ProcessType", "-string", "Background",
-                     "--", "hack/cc.chlc.batt.plist"
+    # Limit config path to Homebrew prefix.
+    system "plutil", "-insert", "ProgramArguments",
+           "-string", "--config=#{File.join(etc, "batt.json")}", "-append",
```
Comment (carlocab) [inline Formula/b/batt.rb:24]:
```suggestion
           "-string", "--config=#{etc}/batt.json", "-append",
```

File: Formula/b/batt.rb [hunk 3]
```diff
@@ -17,27 +17,27 @@ class Batt < Formula
...
+    system "make", "GOTAGS=brew", "VERSION=v#{version}"
     bin.install "bin/batt"
     prefix.install "hack/cc.chlc.batt.plist"
   end
 
   def caveats
     <<~EOS
-      The service must be running before most of batt's commands will work.
+      1. Be sure to disable macOS's built-in Optimized Battery Charging feature.
+      2. The batt service must be running before most of batt's commands will work.
```
Comment (carlocab) [inline Formula/b/batt.rb:40]: We should probably keep the `caveats` as-is -- this is starting to look more like repeating upstream documentation than anything specific to Homebrew.

**Review by carlocab — COMMENTED** [review]
Thanks for the PR; a few comments.

File: Formula/b/batt.rb (hunk 1 above)
Comment (charlie0129) [reply Formula/b/batt.rb]: Sure

File: Formula/b/batt.rb (hunk 3 above)
Comment (charlie0129) [reply Formula/b/batt.rb]: Sure

**Comment by charlie0129** [comment]
Thank you for your suggestions! I fixed the comments.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**